App/Logic/DataUploadLogic.swift

```diff
@@ -232,22 +266,181 @@ extension Logic.DataUpload {
   }
 }
 
+// MARK: - Dummy traffic
+
+extension Logic.DataUpload {
```

Comment:
I'd move all the logic related to the dummy traffic in a separate file. This should help distinguish the concerns

Reply:
Makes sense, done